src/core/webview/webviewMessageHandler.ts

```diff
@@ -82,10 +87,7 @@ export const webviewMessageHandler = async (provider: ClineProvider, message: We
 			const { apiConfiguration: currentApiConfig } = await provider.getState()
 			getOpenRouterModels(currentApiConfig).then(async (openRouterModels) => {
 				if (Object.keys(openRouterModels).length > 0) {
-					await fs.writeFile(
-						path.join(cacheDir, GlobalFileNames.openRouterModels),
-						JSON.stringify(openRouterModels),
-					)
+					await provider.writeModelsToCache(GlobalFileNames.openRouterModels, openRouterModels)
```

Comment:
One of several instances where we have moved to the new higher-level writeModelsToCache() method, alowing `provider.ensureCacheDirectoryExists()` to become private again.